source/agora/network/NetworkManager.d

```diff
@@ -583,6 +620,36 @@ public class NetworkManager
         return new BanManager(banman_conf, data_dir);
     }
 
+    /// register network addresses into the name registry
+    private void onRegisterName ()
+    {
+        const(Address)[] addresses = this.validator_config.addresses_to_register;
+        if (!addresses.length)
+            addresses = InetUtils.getPublicIPs();
+
+        RegistryPayload payload =
+        {
+            data:
+            {
+                public_key : this.validator_config.key_pair.address,
+                addresses : addresses,
+                seq : time(null)
+            }
+        };
+
+        payload.signPayload(this.validator_config.key_pair.secret);
+
+        try
+        {
+            this.registry_client.putValidator(payload);
+        }
+        catch (Exception ex)
+        {
+            log.info("Couldn't register our address: {}. Trying again later..",
+                ex);
+        }
```

Comment:
When would we get an Exception ? The registry will return a string with an error in case of e.g. invalid signature, but we never check the return value.

Reply:
1. ughh, I forgot to check the return value
2. we would get an exception if it times out, right?